app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,182 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:index, :pending, :cancelled, :paid, :completed, :show]
```

Comment:
You may want to use an :except here since require_login is used on most of the actions.